src/main/java/org/broad/igv/sam/cram/IGVReferenceSource.java

```diff
@@ -50,120 +50,23 @@ public class IGVReferenceSource implements CRAMReferenceSource {
 
     private static Logger log = LogManager.getLogger(IGVReferenceSource.class);
 
-    static ObjectCache<String, byte[]> cachedSequences = new ObjectCache<>(5);
-
-    static GenomeChangeListener genomeChangeListener;
-
-    static HashMap<String, Object> locks = new HashMap<>();
-
     @Override
     public byte[] getReferenceBases(SAMSequenceRecord record, boolean tryNameVariants) {
+        return getReferenceBasesByRegion(record, 0,  record.getSequenceLength());
```

Comment:
In the previous implementation there was some extra stuff done to get the length of the chromosome from the Genome object instead of the SAMSequenceRecord.  I changed that because it seemed that if they were different then there's already a bug somewhere since the CRAM file is expecting this to be the matching reference.  Happy to add it back if I'm missing some detail.

Reply:
The chromosome name will not neccessarily match the chromosome name in the CRAM file,  for example the IGV reference might use "chr1" and cram file "1".   The sequence must match of course but chromosome names do not.  This might be what the "extra stuff" was about,  I will look at it.